compute_sdk/globus_compute_sdk/sdk/client.py

```diff
@@ -575,6 +576,8 @@ def register_function(
             Container UUID from registration with Globus Compute
         description : str
             Description of the file
+        metadata : dict
```

Comment:
I used 'details' when adding info to Result/Task, as there was previous 'error_details'.  I thought 'metadata' was a bit too overloaded.

But we did use 'metadata' previously in endpoints, so wonder if 'metadata' is more appropriate.  Too late to change Result to conform, but I guess different fields should be ok.  (This is just a thought, not a suggestion to change).

Reply:
Good thought, but I'd prefer to match the terminology used for endpoints.